evm_loader/program/src/executor.rs

```diff
@@ -341,10 +361,10 @@ impl<'config, B: Backend> Machine<'config, B> {
         let runtime = evm::Runtime::new(code, input, context, self.executor.config);
 
         self.runtime.push((runtime, CreateReason::Call));
+        Ok(())
     }
 
     pub fn create_begin(&mut self, caller: H160, code: Vec<u8>, gas_limit: u64) -> ProgramResult {
```

Comment:
You must record gas for create transaction.
Also check how gas recorded for internal call and internal create.

Reply:
Well, it's done: https://github.com/neonlabsorg/evm/blob/5b72bc771ef85750ad7fff056922a137592b38ea/gasometer/src/lib.rs#L175

As to StackExecutor::call_inner and StackExecutor::create_inner, both look correct for me. Let's review them together to double-check.

Reply:
Added gas recording in all 4 cases: call_begin, create_begin, call & create. Restored the logic from the mainstream rust-blockchain/evm.